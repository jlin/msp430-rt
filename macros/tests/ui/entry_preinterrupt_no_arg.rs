#![no_main]

use msp430_rt_macros::entry;

fn init() {}

#[entry(interrupt_enable(pre_interrupt = init))]
fn main() -> ! {
    loop {}
}
