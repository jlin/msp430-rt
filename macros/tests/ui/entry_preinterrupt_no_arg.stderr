error[E0061]: this function takes 0 arguments but 1 argument was supplied
 --> tests/ui/entry_preinterrupt_no_arg.rs:7:42
  |
7 | #[entry(interrupt_enable(pre_interrupt = init))]
  | -----------------------------------------^^^^--- unexpected argument of type `CriticalSection<'_>`
  |
note: function defined here
 --> tests/ui/entry_preinterrupt_no_arg.rs:5:4
  |
5 | fn init() {}
  |    ^^^^
